<launch>
					<arg name="DEBUG" default="false"/>
					<arg name="uav_name" default="$(optenv UAV_NAME uav1)"/>
					<arg name="gui" default="true"/>
					<arg name="useCuda" default="false"/>
					<arg name="method" default="3"/>
                    <arg name="camnum" default="25001879"/>
					<arg name="SamplePointSize" default="80"/>
					<arg name="ScanRadius" default="4"/>
					<arg name="StepSize" default="0"/>
					<arg name="NumberOfBins" default="20"/>
					<arg name="FrameSize" default="480"/>
					<arg name="publish" default="true"/>
					<arg name="useOdom" default="false"/>
					<arg name="cameraRotated" default="false"/>


				<group ns="$(arg uav_name)">

					<node name="teraranger" pkg="terarangerone" type="terarangerone_node" output="screen" respawn="true">
                        <remap from="terarangerone" to="~range"/>	
			            <!--param name="portname" value="/dev/teraranger" /-->
                        <param name="Mode" value="2"/>
					</node>
					

					<node pkg="bluefox2" type="bluefox2_single_node" name="mv_$(arg camnum)" output="screen">
						<param name="identifier" type="string" value="$(arg camnum)"/>
                        <param name="frame_id" type="string" value="mv_$(arg camnum)"/>
                        <param name="camera_name" type="string" value="mv_$(arg camnum)"/>
                        <param name="calib_url" type="string" value="file://${ROS_HOME}/camera_info/mv_$(arg camnum).yaml"/>
                        <param name="fps" type="double" value="170"/>
                        <param name="idpf" type="int" value="0"/>
                        <param name="aec" type="bool" value="false"/>
                        <param name="expose_us" type="int" value="10000"/>
                        <param name="agc" type="bool" value="false"/>
                        <param name="gain_db" type="double" value="0.0"/>
                        <param name="cbm" type="int" value="0"/>
                        <param name="ctm" type="int" value="1"/>
                        <param name="dcfm" type="int" value="0"/>
                        <param name="hdr" type="bool" value="false"/>
                        <param name="wbp" type="int" value="-1"/>
                        <param name="request" type="int" value="0"/>
                        <param name="mm" type="int" value="0"/>
                        <param name="image_raw/compressed/jpeg_quality" type="int" value="8"/>
                    </node>

		
	  	            <node name="optic_flow" pkg="optic_flow" type="optic_flow_node" output="screen" >
				        <param name="DEBUG"		type="bool" value="$(arg DEBUG)"/> 
				        <param name="gui"             type="bool" value="$(arg gui)"/>
                        <param name="FrameSize"       type="int" value="$(arg FrameSize)"/>
                        <param name="SamplePointSize" type="int" value="$(arg SamplePointSize)"/>
                        <param name="ScanRadius"    type="int" value="$(arg ScanRadius)"/>
                        <param name="StepSize"	type="int" value="$(arg StepSize)"/>
                        <rosparam file="$(find optic_flow)/yaml/ost.yaml"/>
                        <rosparam file="$(find optic_flow)/yaml/cam_rot.yaml"/>
				        <param name="useCuda"		type="bool" value="$(arg useCuda)"/>
                        <param name="method"	type="int" value="$(arg method)"/>
                        <param name="NumberOfBins"	type="int" value="$(arg NumberOfBins)"/>
                        <param name="publish"         type="bool" value="$(arg publish)"/>
                        <param name="ScaleFactor"	type="int" value="1"/>
				        <param name="CameraImagePath"    type="string" value="/$(arg uav_name)/mv_$(arg camnum)/image_raw"/>
                        <param name="RangerDataPath"    type="string" value="/$(arg uav_name)/terarangerone/range"/>
                        <param name="CameraImageCompressed"    type="bool" value="false"/>
                        <param name="cameraRotated"    type="bool" value="$(arg cameraRotated)"/>
    	                <param name="useOdom"           type="bool" value="$(arg useOdom)"/>
	  	            </node>
	            </group>
</launch>
